### PR TITLE
Update typescript definition to latest docs

### PR DESCRIPTION
**Pull request by Sebazzz**
This updates the typescript definitions to the latest from the official docs. In addition, it now defines the format required for routes.

### CODE REVIEW

File: packages/router5/index.d.ts [hunk 1]
```diff
@@ -56,68 +57,332 @@ declare module "router5" {
...
+     * 
+     * @param rootPath The root node path
+     */
+    setRootPath(rootPath : string) : void;
   }
 
+  /**
+   * The result can be synchronous (returning a boolean) or asynchronous (returning a promise or calling done(err, result))
+   */
+  export type RouterActivationHandler = (tostring: State, fromState: State) => boolean | Promise<boolean>;
```
[the remark this answers is not in the document]
Comment (Sebazzz) [reply packages/router5/index.d.ts]: Which errors do you refer to?

[the remark this answers is not in the document]
Comment (troch) [reply packages/router5/index.d.ts]: A rejected promise can return a custom error object

[the remark this answers is not in the document]
Comment (Sebazzz) [reply packages/router5/index.d.ts]: I don't think Typescript types the rejected value of a promise. Regarding your other remark: I've pushed an additional changeset.

File: packages/router5/index.d.ts [hunk 2]
```diff
@@ -56,68 +57,332 @@ declare module "router5" {
...
+
+    /**
+     * if specified, the router will transition to the forwarded route instead. It is useful for defaulting to a child route, or having multiple paths pointing to the same route.
+     */
+    forwardTo?: string;
+
+    /**
+     * a method to control whether or not the route node can be activated
+     */
+    canActivate?: (router: Router) => RouterActivationHandler;
```
Comment (troch) [inline packages/router5/index.d.ts]: Can you make this `RouterActivationHandler`?

Comment (Sebazzz) [reply packages/router5/index.d.ts]: Based on your [docs](http://router5.github.io/docs/preventing-navigation.html) the function accepts a router and returns a callback. Therefore this definition should be correct.

    const canActivate = (router) => (toState, fromState) => {
         return true;
    }

    router.canActivate('admin', canActivate);

**Comment by Sebazzz** [comment]
I've updated the pull request.

**Comment by troch** [comment]
Thanks, will merge and release later today

**Pull request merged**